<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Microhaplotype Constructor and Visualizer • microhaplot</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="Microhaplotype Constructor and Visualizer">
<meta property="og:description" content="A downstream bioinformatics tool to construct and assist 
    curation of microhaplotypes from short read sequences.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">microhaplot</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.0.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="articles/microhaplot-data-prep.html">An Introduction to microhaplot data preparation</a>
    </li>
    <li>
      <a href="articles/microhaplot-walkthrough.html">microhaplot walkthrough</a>
    </li>
    <li>
      <a href="articles/troubleshoot.html">Package Troubleshoot</a>
    </li>
  </ul>
</li>
<li>
  <a href="news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/ngthomas/microhaplot/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">
<div id="microhaplot" class="section level1">
<div class="page-header"><h1 class="hasAnchor">
<a href="#microhaplot" class="anchor"></a>microhaplot</h1></div>
<p><code>microhaplot</code> generates visual summaries of microhaplotypes found in short read alignments. All you need are alignment SAM files and a variant call VCF file. (The latter tells <code>microhaplot</code> which SNPs to include into microhaplotypes). It was designed for extracting and visualized haplotypes from high-quality amplicon sequencing data. We have used it extensively to process amplicon sequencing data (with 100 to 500 amplicons) from rockfish and Chinook salmon, generated on an Illumina MiSeq sequencer. It should be extensible to sequences from capture arrays, like RAPTURE data.</p>
<p>This software exists as an R package <code>microhaplot</code> that includes within it the code to set up and establish an Rstudio/Shiny server to visualize and manipulate the data. There are two key steps in the <code>microhaplot</code> workflow:</p>
<ol>
<li><p>The first step is to summarize alignment and variant (SNP) data into a single data frame that is easily operated upon. This is done using the function <code><a href="reference/prepHaplotFiles.html">microhaplot::prepHaplotFiles</a></code>. You must supply a VCF file that includes variants that you are interested in extracting, and as many SAM files (one for each individual) that you want to extract read information from at each of the variants. The function <code><a href="reference/prepHaplotFiles.html">microhaplot::prepHaplotFiles</a></code> makes a call to PERL to parse the CIGAR strings in the SAM files to extract the variant information at each read and store this information into a data frame which gets saved with the installed Shiny app (see below) for later use. Depending on the size of the data set, this can take a few minutes.</p></li>
<li><p>The second step is to run the microhaplot Shiny app to visualize the sequence information, call genotypes using simple read-depth based filtering criteria, and curate the loci. microhaplot is suitable for quick assessment and quality control of haplotypes generated from library runs. Plot summaries include read depth, fraction of callable haplotypes, Hardy-Weinberg equilibrium plots, and more.</p></li>
</ol>
<p>See the <strong>Example Data</strong> section to learn about how to run each of these steps on the example data that are provided with the package.</p>
<div id="installation-and-quick-start" class="section level3">
<h3 class="hasAnchor">
<a href="#installation-and-quick-start" class="anchor"></a>Installation and Quick Start</h3>
<div id="required-perl-dependencies" class="section level4">
<h4 class="hasAnchor">
<a href="#required-perl-dependencies" class="anchor"></a>required Perl dependencies:</h4>
<p>You need to have Perl (version &gt;5.014) installed in your OS in order to run Microhaplot.<br>
For Window users, we recommend install it via <a href="http://strawberryperl.com/" class="uri">http://strawberryperl.com/</a>.<br>
For Mac and Linux users, Perl can be downloaded from <a href="https://www.perl.org/get.html" class="uri">https://www.perl.org/get.html</a></p>
<p>You can either clone the repository and build the <code>microhaplot</code> package yourself, or, more easily, you can install it using <a href="https://github.com/hadley/devtools">devtools</a>. You can get <code>devtools</code> by <code><a href="https://rdrr.io/r/utils/install.packages.html">install.packages("devtools")</a></code>.</p>
<p><strong>To mac user: remember to install <a href="https://www.xquartz.org/">XQuartz</a>, when upgrading your macOS to a new major version.</strong></p>
<p>Once you have <code>devtools</code> available in R, you can get <code>microhaplot</code> this way:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">devtools</span><span class="fu">::</span><span class="fu">install_github</span><span class="op">(</span><span class="st">"ngthomas/microhaplot"</span>, build_vignettes <span class="op">=</span> <span class="cn">TRUE</span>, build_opts <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"--no-resave-data"</span>, <span class="st">"--no-manual"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p>Once you have installed the <code>microhaplot</code> R package with devtools there you need to use the <code>microhaplot::mvHaplotype</code> to establish the microhaplot Shiny App in a convenient location on your system. The following line creates the directory <code>Shiny</code> in my home directory and then within that it creates the directory <code>microhaplot</code> and fills it with the Shiny app as well as the example data that go along with that.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">microhaplot</span><span class="fu">::</span><span class="fu"><a href="reference/mvShinyHaplot.html">mvShinyHaplot</a></span><span class="op">(</span><span class="st">"~/Shiny"</span><span class="op">)</span> <span class="co"># provide a directory path to host the microhaplot app</span></code></pre></div>
<p>To start familiarizing yourself with microhaplot using the provided example data. We recommend going through our first vignette. Call it up with:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/browseVignettes.html">browseVignettes</a></span><span class="op">(</span><span class="st">"microhaplot"</span><span class="op">)</span></code></pre></div>
<p>and check out <code>microhaplot-walkthrough</code>.</p>
<p>Now, having done that, we can launch Shiny microhaplot on the example data:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/ngthomas/microhaplot">microhaplot</a></span><span class="op">)</span>
<span class="va">app.path</span> <span class="op">&lt;-</span> <span class="st">"~/Shiny/microhaplot"</span>
<span class="fu"><a href="reference/runShinyHaplot.html">runShinyHaplot</a></span><span class="op">(</span><span class="va">app.path</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div id="quick-guide-to-use-microhaplot-to-parse-out-sam-and-vcf-files" class="section level3">
<h3 class="hasAnchor">
<a href="#quick-guide-to-use-microhaplot-to-parse-out-sam-and-vcf-files" class="anchor"></a>Quick Guide to use microhaplot to parse out SAM and VCF files</h3>
<p>This microhaplot package comes with a small customized sample data drawn from an actual run of short read sequencing run on Rockfish species. The sample data contains sequences of eight genomic loci for four populations of five individuals each, with a total of twenty individuals.</p>
<p>First you need to create a tab-separate <strong>label</strong> file with 3 info columns: path to SAM file name, individual ID, and group label (in this particular order). If you do not want assign any group label for the individuals, you can just leave it as “NA”. It is recommended that you have all of the SAM files under one directory to make this labeling task easier.</p>
<p>The <code>label</code> file looks like this:</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode txt"><code class="sourceCode default"><span id="cb5-1"><a href="#cb5-1"></a>s6.sam  s6      copper</span>
<span id="cb5-2"><a href="#cb5-2"></a>s11.sam s11     copper</span>
<span id="cb5-3"><a href="#cb5-3"></a>s13.sam s13     gold</span>
<span id="cb5-4"><a href="#cb5-4"></a>s14.sam s14     kelp</span>
<span id="cb5-5"><a href="#cb5-5"></a>s18.sam s18     gold</span></code></pre></div>
<p>Once you have the label file in place, you can run <code>prepHaplotFiles</code>, a R function that generates tables of microhaplotype, by providing the following:</p>
<ul>
<li>a label to display in haPLOType</li>
<li>path to the directory with all SAM files</li>
<li>path to the <code>label</code> file you just created</li>
<li>path to the VCF file<br>
</li>
<li>optional number of threads (for non-Windows user); recommend 2 * # of processors</li>
</ul>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/ngthomas/microhaplot">microhaplot</a></span><span class="op">)</span>

<span class="co"># to access package sample case study dataset of rockfish</span>
<span class="va">run.label</span> <span class="op">&lt;-</span> <span class="st">"sebastes"</span>

<span class="va">sam.path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html">tempdir</a></span><span class="op">(</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/untar.html">untar</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>,
                  <span class="st">"sebastes_sam.tar.gz"</span>,
                  package<span class="op">=</span><span class="st">"microhaplot"</span><span class="op">)</span>,
      exdir <span class="op">=</span> <span class="va">sam.path</span><span class="op">)</span>
      
<span class="va">label.path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">sam.path</span>, <span class="st">"label.txt"</span><span class="op">)</span>
<span class="va">vcf.path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">sam.path</span>, <span class="st">"sebastes.vcf"</span><span class="op">)</span>
<span class="va">out.path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html">tempdir</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">app.path</span> <span class="op">&lt;-</span> <span class="st">"~/Shiny/microhaplot"</span>

<span class="co"># for your dataset: customize the following paths</span>
<span class="co"># sam.path &lt;- "~/microhaplot/extdata/"</span>
<span class="co"># label.path &lt;- "~/microhaplot/extdata/label.txt"</span>
<span class="co"># vcf.path &lt;- "~/microhaplot/extdata/sebastes.vcf"</span>
<span class="co"># app.path &lt;- "~/Shiny/microhaplot"</span>

<span class="va">haplo.read.tbl</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/prepHaplotFiles.html">prepHaplotFiles</a></span><span class="op">(</span>run.label <span class="op">=</span> <span class="va">run.label</span>,
                            sam.path <span class="op">=</span> <span class="va">sam.path</span>,
                            out.path <span class="op">=</span> <span class="va">out.path</span>,
                            label.path <span class="op">=</span> <span class="va">label.path</span>,
                            vcf.path <span class="op">=</span> <span class="va">vcf.path</span>,
                            app.path <span class="op">=</span> <span class="va">app.path</span>,
                            n.jobs <span class="op">=</span> <span class="fl">4</span><span class="op">)</span> <span class="co"># assume running on dual core</span>
                            

<span class="fu"><a href="reference/runShinyHaplot.html">runShinyHaplot</a></span><span class="op">(</span><span class="va">app.path</span><span class="op">)</span></code></pre></div>
</div>
<div id="suggestions" class="section level3">
<h3 class="hasAnchor">
<a href="#suggestions" class="anchor"></a>Suggestions</h3>
<ul>
<li>SAM files: For pair-ended experiment, both directional reads should be flashed into one.</li>
</ul>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="links">
<h2>Links</h2>
<ul class="list-unstyled">
<li>Download from CRAN at <br><a href="https://cloud.r-project.org/package=microhaplot">https://​cloud.r-project.org/​package=microhaplot</a>
</li>
<li>Browse source code at <br><a href="https://github.com/ngthomas/microhaplot/">https://​github.com/​ngthomas/​microhaplot/​</a>
</li>
<li>Report a bug at <br><a href="https://github.com/ngthomas/microhaplot/issues">https://​github.com/​ngthomas/​microhaplot/​issues</a>
</li>
</ul>
</div>
<div class="license">
<h2>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></small></li>
</ul>
</div>
<div class="developers">
<h2>Developers</h2>
<ul class="list-unstyled">
<li>Thomas Ng <br><small class="roles"> Author, maintainer </small>  </li>
</ul>
</div>

  </div>
</div>


      <footer><div class="copyright">
  <p>Developed by Thomas Ng.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
