<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Extracts haplotype from alignment reads. — prepHaplotFiles • microhaplot</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous" />


<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Extracts haplotype from alignment reads. — prepHaplotFiles" />
<meta property="og:description" content="The function microhaplot extracts haplotype from sequence alignment files through perl script hapture and returns a summary table of the read depth and read quality associate with haplotype." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">microhaplot</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.0.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/microhaplot-data-prep.html">An Introduction to microhaplot data preparation</a>
    </li>
    <li>
      <a href="../articles/microhaplot-walkthrough.html">microhaplot walkthrough</a>
    </li>
    <li>
      <a href="../articles/troubleshoot.html">Package Troubleshoot</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/ngthomas/microhaplot/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Extracts haplotype from alignment reads.</h1>
    <small class="dont-index">Source: <a href='https://github.com/ngthomas/microhaplot/blob/master/R/runHaplot.R'><code>R/runHaplot.R</code></a></small>
    <div class="hidden name"><code>prepHaplotFiles.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>The function <code>microhaplot</code> extracts haplotype from sequence alignment files through perl script <code>hapture</code> and returns a summary table of the read depth and read quality associate with haplotype.</p>
    </div>

    <pre class="usage"><span class='fu'>prepHaplotFiles</span><span class='op'>(</span>
  <span class='va'>run.label</span>,
  <span class='va'>sam.path</span>,
  <span class='va'>label.path</span>,
  <span class='va'>vcf.path</span>,
  out.path <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/tempfile.html'>tempdir</a></span><span class='op'>(</span><span class='op'>)</span>,
  add.filter <span class='op'>=</span> <span class='cn'>FALSE</span>,
  app.path <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/tempfile.html'>tempdir</a></span><span class='op'>(</span><span class='op'>)</span>,
  n.jobs <span class='op'>=</span> <span class='fl'>1</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>run.label</th>
      <td><p>character vector. Run label to be used to display in haPLOType. Required</p></td>
    </tr>
    <tr>
      <th>sam.path</th>
      <td><p>string. Directory path folder containing all sequence alignment files (SAM). Required</p></td>
    </tr>
    <tr>
      <th>label.path</th>
      <td><p>string. Label file path. This customized label file is a tab-separate file that contains entries of SAM file name, individual ID, and group label. Required</p></td>
    </tr>
    <tr>
      <th>vcf.path</th>
      <td><p>string. VCF file path. Required</p></td>
    </tr>
    <tr>
      <th>out.path</th>
      <td><p>string. Optional. If not specified, the intermediate files are created under <code>TEMPDIR</code>, with the assumption that directory is granted for written permission.</p></td>
    </tr>
    <tr>
      <th>add.filter</th>
      <td><p>boolean. Optional. If true, this removes any haplotype with unknown and deletion alignment characters i.e. "*" and "_", removes any locus with large number of haplotypes ( # &gt; 40) , and remove any locus with fewer than half of the total individuals.</p></td>
    </tr>
    <tr>
      <th>app.path</th>
      <td><p>string. Path to shiny haPLOType app. Optional. If not specified, the path is default to <code>TEMPDIR</code>.</p></td>
    </tr>
    <tr>
      <th>n.jobs</th>
      <td><p>positive integer. Number of SAM files to be parallel processed. Optional. This multithread is only available for non Window OS. Recommend two times the number of processors/core.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>This function returns a dataframe of 9 columns i.e group, id, locus, haplotype, depth, sum of Phred score, max of Phred score, allele balance and haplotype rank from highest to lowest read depth. This dataframe will also be saved in <code>out.path</code>.</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'>
<span class='va'>run.label</span> <span class='op'>&lt;-</span> <span class='st'>"sebastes"</span>

<span class='va'>sam.path</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/tempfile.html'>tempdir</a></span><span class='op'>(</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/utils/untar.html'>untar</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/system.file.html'>system.file</a></span><span class='op'>(</span><span class='st'>"extdata"</span>,
                  <span class='st'>"sebastes_sam.tar.gz"</span>,
                  package<span class='op'>=</span><span class='st'>"microhaplot"</span><span class='op'>)</span>,
      exdir <span class='op'>=</span> <span class='va'>sam.path</span><span class='op'>)</span>


<span class='va'>label.path</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/file.path.html'>file.path</a></span><span class='op'>(</span><span class='va'>sam.path</span>, <span class='st'>"label.txt"</span><span class='op'>)</span>
<span class='va'>vcf.path</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/file.path.html'>file.path</a></span><span class='op'>(</span><span class='va'>sam.path</span>, <span class='st'>"sebastes.vcf"</span><span class='op'>)</span>

<span class='fu'><a href='mvShinyHaplot.html'>mvShinyHaplot</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/tempfile.html'>tempdir</a></span><span class='op'>(</span><span class='op'>)</span><span class='op'>)</span>
</div><div class='output co'>#&gt; [1] TRUE</div><div class='input'><span class='va'>app.path</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/file.path.html'>file.path</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/tempfile.html'>tempdir</a></span><span class='op'>(</span><span class='op'>)</span>, <span class='st'>"microhaplot"</span><span class='op'>)</span>

<span class='co'># retrieve system Perl version number</span>
<span class='va'>perl.version</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/numeric.html'>as.numeric</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/system.html'>system</a></span><span class='op'>(</span><span class='st'>'perl -e "print $];"'</span>, intern<span class='op'>=</span><span class='cn'>TRUE</span><span class='op'>)</span><span class='op'>)</span>

<span class='kw'>if</span> <span class='op'>(</span><span class='va'>perl.version</span> <span class='op'>&gt;=</span> <span class='fl'>5.014</span><span class='op'>)</span> <span class='op'>{</span>
<span class='va'>haplo.read.tbl</span> <span class='op'>&lt;-</span> <span class='fu'>prepHaplotFiles</span><span class='op'>(</span>run.label <span class='op'>=</span> <span class='va'>run.label</span>,
                            sam.path <span class='op'>=</span> <span class='va'>sam.path</span>,
                            out.path <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/tempfile.html'>tempdir</a></span><span class='op'>(</span><span class='op'>)</span>,
                            label.path <span class='op'>=</span> <span class='va'>label.path</span>,
                            vcf.path <span class='op'>=</span> <span class='va'>vcf.path</span>,
                            app.path <span class='op'>=</span> <span class='va'>app.path</span><span class='op'>)</span>
<span class='op'>}</span><span class='kw'>else</span> <span class='op'>{</span>
<span class='fu'><a href='https://rdrr.io/r/base/message.html'>message</a></span><span class='op'>(</span><span class='st'>"Perl version is outdated. Must &gt;= 5.014."</span><span class='op'>)</span><span class='op'>}</span>
</div><div class='output co'>#&gt; <span class='message'>Perl is found in system</span></div><div class='output co'>#&gt; <span class='message'>...running Hapture.pl to extract haplotype information (takes a while)...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: `tbl_df()` was deprecated in dplyr 1.0.0.</span>
#&gt; <span class='warning'>Please use `tibble::as_tibble()` instead.</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>...Prepping rds file : /var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T//RtmpIqmMQW/sebastes.rds</span></div><div class='output co'>#&gt; <span class='message'>RDS file: copied into shiny directory: /var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T//RtmpIqmMQW/microhaplot/sebastes*.rds </span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>Run the following to open shiny app: runShinyHaplot("/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T//RtmpIqmMQW/microhaplot")</span></div><div class='input'>
</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Thomas Ng.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


