<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>microhaplot walkthrough • microhaplot</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="microhaplot walkthrough">
<meta property="og:description" content="microhaplot">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">microhaplot</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.0.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/microhaplot-data-prep.html">An Introduction to microhaplot data preparation</a>
    </li>
    <li>
      <a href="../articles/microhaplot-walkthrough.html">microhaplot walkthrough</a>
    </li>
    <li>
      <a href="../articles/troubleshoot.html">Package Troubleshoot</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/ngthomas/microhaplot/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="microhaplot-walkthrough_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>microhaplot walkthrough</h1>
                        <h4 class="author">Thomas Ng and Eric C. Anderson</h4>
            
            <h4 class="date">2021-06-23</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/ngthomas/microhaplot/blob/master/vignettes/microhaplot-walkthrough.Rmd"><code>vignettes/microhaplot-walkthrough.Rmd</code></a></small>
      <div class="hidden name"><code>microhaplot-walkthrough.Rmd</code></div>

    </div>

    
    
<p><code>microhaplot</code> comes pre-loaded with two small, real, example data sets, <code>fish1.rds</code> and <code>fish2.rds</code>, to get you started learning about its functionality. This vignette will walk you through the process of viewing these example data sets and will introduce you to many of the features of <code>microhaplot</code>. Each data set consists of microhaplotypes derived from amplicon sequencing of a marine species, and they represent subsets of data that have been previously employed in full analyses.</p>
<p>The data that we will be exploring will be available after installation of the R package and the Shiny app. Once the R package <code>microhaplot</code> is installed, the Shiny app can be installed like this:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># provide a directory path to host the microhaplot Shiny app</span>
<span class="va">shiny_dir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html">tempdir</a></span><span class="op">(</span><span class="op">)</span>
<span class="fu">microhaplot</span><span class="fu">::</span><span class="fu"><a href="../reference/mvShinyHaplot.html">mvShinyHaplot</a></span><span class="op">(</span><span class="va">shiny_dir</span><span class="op">)</span></code></pre></div>
<p>where <code><a href="https://rdrr.io/r/base/tempfile.html">tempdir()</a></code> should be replaced by a location on your hard drive where you want the Shiny app installed.</p>
<p>Once it is installed, the Shiny app is run like this:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">app.path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">shiny_dir</span>, <span class="st">"microhaplot"</span><span class="op">)</span>
<span class="fu">microhaplot</span><span class="fu">::</span><span class="fu"><a href="../reference/runShinyHaplot.html">runShinyHaplot</a></span><span class="op">(</span><span class="va">app.path</span><span class="op">)</span></code></pre></div>
<div id="about-the-example-data" class="section level1">
<h1 class="hasAnchor">
<a href="#about-the-example-data" class="anchor"></a>About the example data</h1>
<p>The data set in <code>fish1.rds</code> contains microhaplotypes from 22 amplicons in 436 individuals of one population group. <code>fish2.rds</code> consists of data from 14 amplicons in 288 individuals from two populations. The actual sequencing runs that produced the data contained many more loci, but we have subsampled the original data sets in order to make the data portable. The motivation behind supplying two datasets is to show that, despite both being derived from amplicon-sequencing reads, there is great variation in read depth between the two data sets, and this will affect the reliability of called genotypes.</p>
</div>
<div id="microhaplot-layout" class="section level1">
<h1 class="hasAnchor">
<a href="#microhaplot-layout" class="anchor"></a>Microhaplot layout</h1>
<p>The <code>microhaplot</code> shiny viewer is split into three horizontal “panels”. From top to bottom, they are:</p>
<div id="interaction-tabset-top" class="section level2">
<h2 class="hasAnchor">
<a href="#interaction-tabset-top" class="anchor"></a><strong>Interaction tabset</strong> (top)</h2>
<p>These are the three tabs that live at the top of the page with the light blue background. Any one of the three: “&gt; field selection”, “+ read criteria”, and “+ locus annotation” can be selected. Whichever one is selected determines the content of the panel directly beneath—the user input panel, the panel with the light gray background.</p>
<ol style="list-style-type: lower-alpha">
<li><p><strong>Field selection tab:</strong><br><br><img src="field_selection.png" title="read criteria tab" width="100%"><br><br> If the “&gt; field selection” is selected, then the user input panel will include dropdown boxes that let you choose a group, locus, or individual(s) that you wish to focus on (this choice affects what is shown in different plots and outputs, etc.)<br><br> The top choice in each dropdown box is “ALL” which indicates that results for all (groups, individuals, or loci) are to be portrayed at once. This tab also includes “backwards-and-forwards” buttons, “&lt;” and “&gt;”, that sit immediately to the right of the <strong>Indiv</strong> and the <strong>Locus</strong> dropdown boxes, and which allow you to step backward or forward through the content of the dropdown boxes. These “backwards-and-forwards” buttons are very useful when you want to work your way through each locus, one at a time, to choose those that do (and do not) appear to be reliable enough to include in a panel of amplicons. (<em>Note:</em> the location and relative position of each button might change, depending on the width of the viewing window.)</p></li>
<li>
<p><strong>Read criteria tab:</strong> <br><br><img src="read_criteria_tab.png" title="read criteria tab" width="100%"><br><br> In the “+ read criteria” tab, you are given the choice to define the top <span class="math inline">\(n\)</span> alleles, minimum read depth and minimum allelic ratio. These values specify a series of filters that microhaplotypes must satisfy to pass through the analysis. It is recommend to adjust these filters in a particular order, as follows:</p>
<ol style="list-style-type: lower-roman">
<li><p><code>Top n Alleles</code> defines the maximum number of distinct microhaplotype sequences (in a single individual’s locus) to be kept, starting from the highest-read-depth haplotype and proceeding to haplotypes found in lower read depths. Ideally, the top n alleles should match the ploidy of the species you study. So for a diploid organism, the <em>top n alleles</em> should be set to 2. For a tetraploid, the <em>top n alleles</em> should be set to 4.</p></li>
<li><p><code>Min Total Read Depth</code> or <em>Minimum Total Read Depth</em> defines the minimal total number of reads from the “Top n alleles” required for a locus to pass filter within an individual. It can only take values greater than or equal to 0.<br><br>For example, if <code>Top n Alleles</code> is set to 2, and <code>Min Total Read Depth</code> is set to 15, then if Individual A’s two highest read depth haplotypes at Locus 1 are CCA with 10 reads and GGA with 7 reads, then the two haplotypes retained from Individual A at Locus 1 are CCA and GGA.<br><br>By contrast, if Individual B’s two highest-read-depth haplotypes at Locus 1 are GTC with 6 reads and GGA with 5 reads, then, because the total number of reads of those two haplotypes is 11, which is less then 15, none of Individual B’s haplotypes at Locus 1 will pass this criterion, and, therefore, no genotype will be reported for Individual at Locus 1.</p></li>
<li><p><code>Min Allelic Ratio</code> or <em>Minimum Allelic Ratio</em> defines the minimum allelic ratio that a haplotype needs to pass filter. It must be between 0 and 1. For a particular locus, L, of individual I, the allelic ratio of haplotype/allele X is defined as the ratio of the observed read depth of haplotype/allele X at L in I to the read depth of the haplotype/allele with highest read depth at L in I.<br><br>The <code>Min Allelic Ratio</code> filter removes from consideration those reads of an allele/haplotype that occur in low depth relative to the highest-read-depth allele. It is important to understand that the <em>allelic ratio</em> is a quantity that applies to each distinct haplotype (allele) at a locus within an individual, and it is relative to the haplotype/allele with the highest read depth. Accordingly, the values of allelic ratios you should expect to see in an individual that is truly homozygous or heterozygous are as follows:</p></li>
</ol>
<blockquote>
<p>In a <strong>homozygous</strong> individual, the highest read depth allele should be the one carried in n copies by the n-ploidy individual. Therefore you expect that true allele/haplotype to have a large read depth and an allelic ratio of one while other alleles/haplotypes should have low allele-specific read depth and allelic ratio close to zero.</p>
</blockquote>
<blockquote>
<p>In a truly <strong>heterozygous</strong> diploid, the two alleles/haplotypes carried by the individual will, ideally, each have a large and similar number of reads, which means that their allelic ratios should be at or near one.</p>
</blockquote>
<p>Once you set up these three criteria, you <em>apply</em> them to see their effects on the currently displayed data visualization with the <em>apply</em> button. You can also <em>save</em> these criteria if you want the values to “stick”, by clicking the “save” button.</p>
<ol start="4" style="list-style-type: lower-roman">
<li>
<em>Additional options for applying and saving</em>: When <code>ALL</code> is chosen from the dropdown box in the “&gt; field selection” tab, additional filtering options will be available at the right hand side of the “+ read criteria” tab. Details for each of the options are displayed if you hover your pointer over it:
<ul>
<li>
<code>overrides only as min baseline</code>: when selected and <em>apply</em> or <em>save</em> is chosen, the selected values for all three of the aforementioned criteria (<code>Top n Alleles</code>, <code>Min Total Read Depth</code>, and <code>Min Allelic Ratio</code>) will replace the current values of the criteria for every locus in which the saved value is greater than the currently selected values. This is useful when you want to set a minimal baseline for every single locus. For example, if minimal total read depth for loci A and B are set set to 5 and 11, respectively, and the value for <code>ALL</code> has been set to 10, then the minimal total read depth for locus A will be updated to 10 when this option is selected and <em>apply</em> or <em>save</em> is chosen.</li>
<li>
<code>overrides all params</code>: This option, when selected and <em>apply</em> of <em>save</em> is chosen, will set <code>Top n Alleles</code>, <code>Min Total Read Depth</code>, and <code>Min Allelic Ratio</code>, to be whatever the currently selected values are. Be wary that there’s no way to undo this process once this option is selected and applied. For example, if minimal total read depth for locus A, B, is set to 5 and 11, and the value for <code>ALL</code> has been set to 10, then the minimal total read depth for locus A and B will both be updated to 10 when this option is selected and <em>apply</em> or <em>save</em> is chosen.</li>
<li>
<code>relies on local locus param.</code>: It is very helfpul to select <code>ALL</code> loci for viewing in broad summaries such as those in the <em>Summary-&gt;Global Scope</em> panel described below. When choosing these views, one might want to preserve each locus’s specific filtering criteria (as saved using the steps above) rather than the criteria applied to <code>ALL</code> loci. In order to effect this, the <code>relies on local locus param.</code> box should be checked and, after that, the <em>apply</em> button should be activated (it is important to hit the <em>apply</em> button every time you switch back to <code>ALL</code> loci.</li>
</ul>
</li>
</ol>
<p>Understanding how these options work can take a little practice. We recommend that the user spend some time playing around with the various options while viewing the effects on the “locus annotation” output of the Table panel (see the Table section under the Panel Choice Tabset section below.)</p>
</li>
<li><p><strong>Locus annotation tab:</strong><br><br><img src="locus_annotation.png" title="read criteria tab" width="100%"><br><br> This tab is for users to keep track of whether a certain locus should be retained based on its quality or other attributes. This is most useful when a particular locus has been chosen under the “&gt; field selection” panel. There is also a small textbox that lets you insert any comments you would like to record about a particular locus. You must press the <strong>save button</strong> to submit this information about each locus. All of this locus-annotation information can be retrieved under the table selection for “locus annotation” under the <code>Table</code> tab (see below).</p></li>
</ol>
<blockquote>
<p><strong>Ways to access saved criteria values and locus annotations in file format:</strong> All of the saved parameters such as <code>Min Total Read Depth</code>, as well as the locus annotations, are stored in a data frame in memory and are read in and saved as serialized R objects in an RDS file in the same directory path as the shiny app. The name of this file has the suffix <code>_annotate.rds</code>. For example, for <code>fish1.rds</code> data, the file name is <code>fish1_annotate.rds</code>. That file can be used to retrieve the data. A better way to retrieve the data is by downloading the “locus annotation” table from the <code>Table</code> tab at the bottom pane. For the courageous, you can set the filters and annotations elsewhere, save those into an RDS file named appropriately (with the <code>_annotate.rds</code> suffix) and then insert that file into your microhaplot analysis by copying it to the directory path where the shiny app resides. This is not recommended for the faint of heart, as bungling the format of the data frame (or the exact locus names) in that RDS file could have unforeseen, unexpected, or even catastrophic consequences…</p>
</blockquote>
</div>
<div id="main-body-panel-center" class="section level2">
<h2 class="hasAnchor">
<a href="#main-body-panel-center" class="anchor"></a><strong>Main body panel</strong> (center)</h2>
<p>This panel occupies most of microhaplot’s screen real estate. It is where the actual plots, tables, and other results are rendered. What gets shown here depends on which main body panel has been chosen - located right at the bottom. That choice is made using the <em>main panel choice tabset</em> described below.</p>
</div>
<div id="panel-choice-tabset-bottom" class="section level2">
<h2 class="hasAnchor">
<a href="#panel-choice-tabset-bottom" class="anchor"></a><strong>Panel choice tabset</strong> (bottom)</h2>
<p>This tabset is always down at the bottom of the screen, and the choices on this tabset allow you to choose the contents to be displayed in the main body panel. It looks like this:<br><br><img src="main_tab.png" title="main tab" width="100%"><br><br> In the following, we will describe all these different main panels, but, since they are rather large, we will not include any pictures of those panels. It is recommended that the reader follow along with this vignette using the example data in microhaplot. That will help a lot.</p>
<ol style="list-style-type: lower-alpha">
<li><p><em>Data Set:</em> provides a dropdown box in the main panel that you use to choose which data set you want to explore. The choices in the dropdown box are taken from the <code>.rds</code> files that are in the <code>app.path</code> directory described above. When you start microhaplot after a clean install, the default dropdown choice is <code>fish1.rds</code>, and the other possible choice is <code>fish2.rds</code>.<br></p></li>
<li><p><em>Summary Info: </em> displays a condensed graphical summary of the characteristics of microhaplotype reads by groups, individuals, or loci. These are particularly useful for overviews when the “ALL” choice is given for individuals and loci in the “&gt; field selection” tab.<br></p></li>
<li>
<p><em>Criteria Cutoff: </em> contains:</p>
<ol style="list-style-type: lower-roman">
<li>
<code>Global Scope</code>: This section displays summaries of the distributions of read depths and allelic balance ratios to assist the user in setting appropriate values for those filter cutoffs. Histograms of read depths and allelic balance ratios are presented for only one locus, or combined across <code>ALL</code> loci, depending on the settings in the “&gt; field selection” tab. If only a single locus is selected, the histogram of read depths and allelic balance ratios are shown for each allele, as well. Looking at the distribution of allele-specific read depths can be very useful for identifying alleles that are linked to variation that interferes with PCR of the amplicon. At the bottom, beneath the graphical histograms, we also provide a tabular summary of the number of alleles and the number of individuals with callable genotypes (given the filters) available at different cutoffs for the minimum total read depth and the minimum allelic ratio.<br>
</li>
<li>
<code>Quality Profiling</code>: This segment displays individuals or loci that may carry more alleles with high read depth than expected, possibly indicating that the sample is contaminated, or that the amplicon targets paralogous or repetitive loci. To highlight those problematic loci or individuals, a <em>per-allele</em> minimum read depth filter is used, rather than a minimum <em>total</em> read depth filter. This per-allele read depth is set using the <code>Min read depth (per haplotype)</code> slider input at the top part of the main panel. The upper plot graphically summarizes the individuals or loci that might have more high-read-depth alleles than expected. The individual list shows a point at 0, 1, 2, …, where each of those values gives the number of alleles above the “Top n” filter criterion that have read depths that pass the allele-specific read depth filter. Having a large point at 0 is good. Points at values greater than 0 record the number of individuals with that many loci having extraneous alleles of high read depth. The “Loci List” provides an analogous summary for loci. The details of all these items can be further examined in the bottom table. The Search box at the upper right of the table is particularly useful for tracking down a specific individual sample or locus in the table output.</li>
</ol>
</li>
<li>
<p><em>Genotype Call: </em> offers a more intimate look and analysis of how genotypes are being called at a single locus, and of the allele specific read-depths within those genotypes.</p>
<ol style="list-style-type: lower-roman">
<li>
<code>AR refinement</code>: This panel displays a scatterplot that shows the relationship between the allele with the highest read depth and the allele with the second-highest read depth. It is most suited to handling data from diploid organisms. This space also provides you the tools (found on the left) to fine-tune allelic ratio parameters (such as defining the maximum allelic ratio for homozygous genotype calls and the minimum allelic ratio for heterozygous genotype calls) so that individuals with questionable allelic balance ratio will not receive called genotypes.<br><br> The lines on the plot are as follows:
<ul>
<li>
<span style="color:purple">Solid Purple Line</span>: This shows the consequence of the `Min Total Read Depth’ filter. It is slanted down and right because it is a criterion on the sum of the read depths of the two highest-read-depth alleles. Points that fall below and to the left of this line appear as X’s because they are no-calls.</li>
<li>
<span style="color:orange">Dashed Orange Line</span>: This line shows where the standard <code>Min Allelic Ratio</code> filter resides. It shows that any point below that line will, by default, be considered a heterozygote because the allele with second-highest read depth has an allele balance ratio that is high enough (i.e. its read depth is close enough to that of the highest read depth allele that it is believed to be a true allele.)</li>
<li>
<span style="background-color:yellow">Solid Yellow Line</span>: This line shows the Min Allelic Ratio for Heterozygotes cutoff associated with the <code>Min AR for Het (yellow)</code> slider input to the left side of the panel. In the beginning of an analysis, it will be set the same as the dashed orange line, but can be tailored for specific loci. For example, some loci might have most heterozygous individuals with an allelic ratio near 1, but a few of them have allelic ratio near 0.6. To exclude those individuals with low-ish allelic ratio, you could set the <code>Min AR for Het (yellow)</code> to 0.7, which will move the yellow line down, and will consider the aberrant heterozygous individuals as no-calls.</li>
<li>
<span style="color:aqua">Solid Aqua Line</span>: This shows the Max Allelic Ratio for Homozygotes cutoff associated with the <code>Max AR for Homoz</code> slider input. Changing this allows you to filter out those individuals that are considered to be homozygotes, but those whose second-highest-read-depth allele is concerningly high.</li>
</ul>
<br> In order to save the Min/Max AR values chosen for heterozygotes or homozygotes, you can choose <em>save</em> from the “+ read criteria” tab.<br><br> Below the scatterplot, you will also find the distribution of read depths and the allelic depth ratios of different haplotype-alleles in relation to whether each alleles was included in a genotype call and whether or not that genotype call was considered homozygous or heterozygous.</li>
<li>
<code>Summaries</code> shows haplotype frequencies within each group, and a visual interactive summary of conformance to Hardy-Weinberg proportions of individual loci, and also portrays the position of these variants within the sequence.</li>
</ol>
</li>
<li><p><em>Table: </em> This panel provides a place to get a number of tabular views (available in the “Select Table:” dropdown box) of the data that are sortable and searchable. The output of each is self-explanatory with a little exploration. This can be very useful when you want to dig into the fine details of the short read data. It also allows you to download filtered and called data and genotypes for later analysis. Note that clicking a column name will sort on that column (purple triangle pointing up or down depending on sort order, which can be changed be clicking it again). To sort on multiple columns, you can shift-click them. The rows can be filtered by those rows matching a certain string but putting that string in the Search box. For this to be most useful it is recommended that all sample and locus names not be simply numeric (i.e. call it <code>Loc_110</code>, instead of <code>110</code>, and <code>Fish_36</code>, instead of simply <code>36</code>). Note that the <code>fish1.rds</code> data set does not adhere to that convention for sample names.</p></li>
<li><p><em>About (microhaplot): </em> provides a little background about microhaplot and how to cite it and whom to contact about it.</p></li>
</ol>
<p>With that background we are ready to start exploring the example data.</p>
</div>
</div>
<div id="mission-one-an-overview-of-loci" class="section level1">
<h1 class="hasAnchor">
<a href="#mission-one-an-overview-of-loci" class="anchor"></a>Mission one: An overview of loci</h1>
<p>The first thing we are going to do is get an overview of the data with no filtering, focusing on different loci. To do this, we choose <em>Summary Info –&gt; By Locus</em>. We also want to make sure that we are starting from the defaults in “&gt;field selection” (ALL group, ALL loci, ALL individuals), and under “+read criteria” you want to ensure that “Minimal Read Depth” and “Minimal Allelic Ratio” are both set to 0. (If you had changed these before, then, once you set them back to 0, you should hit the “apply” button to apply the changes.)</p>
<p>Choosing <em>Summary Info –&gt; By Locus</em> will give you four plots. By default, only the first 15 loci are represented (the reason for this is that if you have many loci, you don’t want to end up waiting for an eternity to have the final plots rendered.) You can change the number of loci shown using the “Display” dropdown menu that lives just beneath the user input panel. You can also choose to go through the loci, 15 at a time by choosing which page to go to using the “Page” selector to the right of the “Display” selector.</p>
<p>The four figures shown in this <code>By Locus summary</code> are, from left to right:</p>
<ol style="list-style-type: decimal">
<li><p>The distribution of the number of haplotypes observed per individual. Recall that these are subject to the current filters, of which there are none, so each haplotype, even if it is only observed at a read depth of 1 read in an individual is counted as observed. This plot is a bubble plot that essentially is giving histogram-like information. The y-axis has different locus on each line. The x-axis is the number of unique haplotypes and the size and shading of the bubbles indicate how many individuals in the data set had the given number of unique haplotypes. It is particularly revealing to focus on locus <code>tag_id_1166</code> (fifth locus in main body panel). Most individuals have a large number of different haplotypes, likely occurring at very low read depths. At any rate, this plot makes it clear that there are a lot of low-read-depth sequences that are superfluous and incorrect.</p></li>
<li><p>Moving to the right is another figure that has loci arranged on the y-axis and number of haplotypes on the x-axis. However, in this case, the number is counting the total number of unique haplotypes observed at the locus when only the two highest-read-depth haplotypes are recorded per individual.</p></li>
<li><p>The third plot is a plot with light blue points indicating the fraction of individuals that have sufficient read depth to allow a genotypes to be called given the current settings of the read depth and allelic ratio filters.<br>
Obviously since there is filtering, every locus is going to have a genotype called at every individual.</p></li>
<li><p>The final plot (the furthest right) just shows violin plots expressing the distribution of read depths at each locus across the individuals. Notice that the x-axis is on a log scale. The plus denotes the median read depth.</p></li>
</ol>
<p>Now that we know what all of these plots mean, we can interpret what we see.</p>
<p>We note that most loci have a read depth greater than 20 at all the individuals. Also there is a huge amount of variation across loci in the number of recorded haplotypes per individual. Since any diploid individual will have at most two distinct haplotypes at a locus, most of these haplotypes are spurious, and we can hope that most of them are at very low read depths and will be filtered out. We will investigate that now.</p>
<div id="lets-apply-some-filters" class="section level2">
<h2 class="hasAnchor">
<a href="#lets-apply-some-filters" class="anchor"></a>Let’s apply some filters</h2>
<p>Here is the moment we have all been waiting for. We have been looking at our data in a totally unfiltered state—effectively regarding a haplotype with 1 read to be on an equal footing with one having, say, 1000 reads in an individual. There are two, simple, main filtering methods in microhaplot. The first is to discard haplotypes from individuals if they occur in fewer than a certain number of reads (The “Minimal Read Depth” criterion). The second is to discard haplotypes from individuals if their read depth is less than a given fraction of the haplotype with the highest read depth at the locus in the individual (The “Minimal Allelic Ratio” criterion). Choose 25 for the Minimal Read Depth and slide the bar to 0.2 for the Minimal Allele Ratio, and then hit “Apply”.</p>
<p>Once those filters take effect we find that most individuals are observed to carry one or two haplotypes at each locus. Some loci have more haplotypes that that observed in some individuals. We will want to check up on those later. Some of the loci clearly did not sequence well. For example <code>tag_id_1511</code>. (hint: look for violin plot w/ skewed far left distribution) To see an expanded view of a few loci, you can highlight (with a click drag) their rows in the left hand figure and then double click those to get a zoomed in view. To zoom back out, to be able to see all the loci, just double click the body of the left figure.</p>
</div>
</div>
<div id="mission-two-an-overview-of-individuals" class="section level1">
<h1 class="hasAnchor">
<a href="#mission-two-an-overview-of-individuals" class="anchor"></a>Mission Two: An overview of individuals</h1>
<p>We can take those same filters we just applied and get a good overview of our sequencing data on an individual basis. Go ahead of choose <em>Summary Info–&gt;By Individual</em>. from the main panel choice tabset on the bottom of the screen. The initial screen that comes up includes only 15 of the individuals. We can change that by choosing “ALL” in the “Display (indiv/pg)” dropdown.</p>
<p>This “By Individuals” summary view of the data provides four different plots. From left to right they are:</p>
<ol style="list-style-type: decimal">
<li><p>The ratio of the haplotype with 2nd highest read depth to the haplotype with highest read depth.<br>
In this plot, the y-axis denotes individuals, the x-axis represents the ratio (it goes from 0 to 1) and each locus is represented as a bubble. The size of the bubble is scaled to some function of read depth, but it is hard to place that on an absolute scale. The main take home message from this plot is that for most individuals at most loci the two highest-read depth haplotypes have fairly similar read depths. For example a ratio of 0.75 indicates that if the haplotype with highest read depth at a locus in an individual was 100 reads, then, the haplotype with second highest read depth was 75 reads.</p></li>
<li><p>The number of haplotypes observed in each individual. This is a bubble plot that expresses the relative number of loci at which 1, 2, 3, etc haplotypes were observed after filtering.</p></li>
<li><p>The number of loci within the individual with sufficient read depth to call a genotype (given the current read depth and allelic ratio filters).</p></li>
<li><p>The distribution of read depths at all the haplotypes that were not filtered out at all the loci in the individual. These are portrayed as violin plots with individual values as dots within them.</p></li>
</ol>
<p>It should be noted that the colors of the bubbles on the left-most graph indicate the different groups that the individuals fall into. This is particularly useful when individuals from different populations or species are in your dataset. In the case of our example data, we have merely assigned half of the individuals to be from “pop1” and the other half from “pop2”.</p>
<div id="spotting-some-dodgy-individuals" class="section level2">
<h2 class="hasAnchor">
<a href="#spotting-some-dodgy-individuals" class="anchor"></a>Spotting some dodgy individuals</h2>
<p>This overview of all the individuals gives us a good chance to spot a few problem fish right off the bat. Just scanning down the fraction of loci that the individuals might be called at, it is clear that s45 and s277 had limited sequencing success. That is clear by looking at the distribution of read depths across loci. Perhaps even more intriguing is individual s288 that seems to have pretty reasonable read depths, but very poor allelic balance.</p>
</div>
<div id="looking-at-the-raw-counts-for-s288" class="section level2">
<h2 class="hasAnchor">
<a href="#looking-at-the-raw-counts-for-s288" class="anchor"></a>Looking at the raw counts for s288</h2>
<p>Another way to look at the data is through the tabular format. If we want to see tabular data for individual s288, choose “Table” from the main panel choice tabset. By default this will show a few lines of the “Observed variants (unfiltered)” table. You can change which table is viewed using the “Select Table” dropdown, but unfiltered observed variants is great for now.</p>
<p>In the search box on the right, enter <code>s288</code> and hit return. Note: search box is not case sensitive. This will filter the table to only those rows that include s288. Once that is done, click on the “locus” column heading to sort entries by locus. Voila! Now you can peruse all the observed variants and their read depths at this individual s288. Where it says “Show 15 entries” on the left hand side just above the table you should change the “15” to “All”. This will give you a single scrollable window of all the entries for s288. Note that the column “Haplo” gives the sequence of the reads at the variable positions. These are essentially the haplotypic states of the reads.</p>
<p>Going through the data for 288, it is clear that there will be some genotypes that are not straightforward to call. For example, at locus <code>tag_id_879</code> there are three different haplotypes with read depths over 50: 128 reads of CCCTCTG, 100 reads of CCCTCAG, and 57 of GCCCCTG. That is clearly going to be difficult to genotype. Perhaps this individual has contaminated DNA or represents two individuals that were barcoded equivalently. It probably should be tossed out of the final data set.</p>
</div>
<div id="raw-count-data-for-a-less-dodgy-individual" class="section level2">
<h2 class="hasAnchor">
<a href="#raw-count-data-for-a-less-dodgy-individual" class="anchor"></a>Raw count data for a less dodgy individual?</h2>
<p>So, the data for s288 was a little disquieting, but recall that was an individual that looked pathological from the very beginning. So, what if we look at the raw count data for an individual that doesn’t look so dodgy? This will be a good exercise to build up some intuition about how genotyping might be done with microhaplotype read data.</p>
<p>Let’s have a closer look at individual s95. From the overview, this is an individual with reasonably clean, abundant reads. We can filter on this individual merely by replacing “s288” in the search box with “s95”. A cursory glance at the results for this individual show us what appears to be an individual that will be very easy to genotype: there are never more than two haplotypes at high read depth but there is always at least one haplotype at high read depth. When two haplotypes are at high read depth, their read depths are largely quite similar.</p>
</div>
</div>
<div id="looking-over-different-loci" class="section level1">
<h1 class="hasAnchor">
<a href="#looking-over-different-loci" class="anchor"></a>Looking over different loci</h1>
<div id="summaries" class="section level2">
<h2 class="hasAnchor">
<a href="#summaries" class="anchor"></a>Summaries</h2>
<p>When we are interested in assessing how well each locus behaves in reporting microhaplotypes, it is time to turn to the “Filter Analysis” panel choice. First, go ahead and choose <em>Criteria Cutoff –&gt; Global Scope</em>. This view gives you histograms of the read depths and allele balance ratios (the filtering choices in effect appear as dashed red lines), and, below that, a tabular perspective on the outcome of different filtering choices on the total number of haplotypes recorded in the data set. The field selection choices that are present are applied here. So, if ALL loci are selected, then the histograms include reads from all the haplotypes at all the loci, and the tabular summary counts up the total number of haplotypes typed and the total number of individuals typed at all loci. If you select just one locus, the results reflect that one locus, and the histogram results are further broken down by different haplotypes within that locus.</p>
<p>With that in mind, choose the first locus in the dropdown menu: <code>Plate_1_A11_Sat_GE820299_consensus</code> under “&gt; field selection” and see how that changes the broad summary. Note that across a broad range of the two filtering options (minimum read depth and minimal allelic balance) there are eight haplotypes in total discovered.</p>
<p>In order to search for individuals or loci that have more high-read-depth alleles that one would want, you can choose <em>Criteria Cutoff –&gt; Quality Profiling</em>. This view shows individuals or loci that have more than <span class="math inline">\(n\)</span> alleles that pass the filters. For example, if you are dealing with diploids, then you would set <span class="math inline">\(n\)</span> to 2 (in the “+read criteria: Top n Alleles” option). Then any individuals or loci that had more than 2 alleles that satisfied the minimum read depth and allele balance criteria would be noted here. This is good way to look out for contaminated samples or loci that amplify paralogous regions.</p>
<p>To see how the inferred haplotypes look in terms of haplotype frequencies, and also how the genotypes look in terms of Hardy-Weinberg equilibrium, you can choose <em>Genotype Call –&gt; Summaries</em>. This view consists of four figures. The first, in the upper left simply shows the frequencies (and the total read depth) of different haplotypes. The plot in the upper right shows the relationship between the observed frequency of different genotypes and the expected frequencies under Hardy-Weinberg equilibrium. The individuals used in creating these summaries depends on which “Group” is chosen. In this case we have “ALL” chosen, and that is fine because the two groups are essentially identical, genetically. However, if we were dealing with groups that were genetically differentiated, we would not want to assess conformance to Hardy-Weinberg proportions of a mixture of those different groups! In such a case it is worthwhile to look at one group at a time.</p>
<p>The expected number of different genotypes is shown by the outlines of circles and the observed number by the filled, colored circles. Green are homozygotes, orange are heterozygotes, and it should be relatively self-explanatory. There is not a scale, but if you click in the center of any of the genotypes with observed (non-zero) counts, you will be told (in the upper left of the panel) what the expected and observed numbers were for that genotype. These plots are not meant to provide a defensible test of departures from Hardy-Weinberg equilibrium, but do allow the user to diagnose loci that are grotesquely far out of Hardy-Weinberg equilibrium.</p>
<p>Below the haplotype frequencies and HW conformance plots you will find a simple bubble plot expressing haplotype frequencies in the different groups. In the case of the example data there are only two different groups and they have very similar allele frequencies. This plot becomes more useful when one is comparing allele frequencies across many different groups.</p>
<p>Finally, you may need to scroll down to see the final figure in this display. It is a representation of the haplotype sequences, their frequencies, and the positions of the variants within them along each amplicon.</p>
</div>
<div id="allele-biplots" class="section level2">
<h2 class="hasAnchor">
<a href="#allele-biplots" class="anchor"></a>Allele Biplots</h2>
<p>Choosing <em>Genotype Call –&gt; AR Refinement</em> takes you to a very informative screen. It is described above in the vignette. Read through the section that describes it and then try playing with the sliders to move the four different lines around the plot and see the effect on whether genotypes get called or not.</p>
<p>Note that you can use the blue lasso tool (upper right corner of the scatter plot) to select a lot of points, whose values will then be revealed in a table below. The red lasso can be used to de-select points and you can check the box “keeps pt selection between loci” to maintain focus on those points as you move from locus to locus. This can be very useful for identifying individuals that show aberrant read depths across multiple loci.</p>
</div>
</div>
<div id="output-of-data" class="section level1">
<h1 class="hasAnchor">
<a href="#output-of-data" class="anchor"></a>Output of data</h1>
<p>We have already witnessed how to use the <em>Table</em> panel to look at the raw data. It is also possible to download filtered, unfiltered, and processed data from this view using the “Download Data” button. This saves output in CSV files.</p>
</div>
<div id="locus-annotation" class="section level1">
<h1 class="hasAnchor">
<a href="#locus-annotation" class="anchor"></a>Locus annotation</h1>
<p>The program can be used to annotate different loci (and assign filtering criteria to them). This annotation can be done using the “+locus annotation” interaction tabset, and the results saved by downloading the locus annotation data from the <em>Outputs</em> panel.</p>
<div id="importing-external-input-data" class="section level3">
<h3 class="hasAnchor">
<a href="#importing-external-input-data" class="anchor"></a>importing external input data</h3>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Thomas Ng.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
